Foundation/src/NotificationCenter.cpp

```diff
@@ -29,6 +29,14 @@ NotificationCenter::NotificationCenter()
 
 NotificationCenter::~NotificationCenter()
 {
+	try
+	{
+		Mutex::ScopedLock lock(_mutex);
+	}
```

Comment:
Since `_mutex` protects `_observers` was it meant that list of observers is cleared while holding the mutex in scoped lock?

Otherwise I do not see the point of having the lock in place.

Reply:
yes, this was reported by clang thread sanitizer #4307 
This fix is not pretty and not even bulletproof but I guess it narrows the race opportunity somewhat; an alternative could be making `_observers` a `SharedPtr`, and then temporarily holding a reference to it in all accessors

Reply:
What about having it like this?

```
NotificationCenter::~NotificationCenter()
{
	try
	{
		Mutex::ScopedLock lock(_mutex);
		for (auto& o: _observers)
			o->disable();

		_observers.clear();
	}
	catch(...)
	{
		poco_unexpected();
	}
}

```

Reply:
yes, that's fine. you can change it and merge this PR